stream/sentinel-intel/src/sentinel_intel_connector/utils.py

```diff
@@ -0,0 +1,188 @@
+
+
+def is_observable(data: dict) -> bool:
+    """
+    Check if data represents a STIX Observable.
+    :param data: Data to check
+    :return: True if data represents a STIX Observable, False otherwise
+    """
+    return data["type"] in OBSERVABLE_TYPES
+
+
+def get_ioc_type(data: dict) -> str | None:
+    """
+    Get valid IOC type for Sentinel from data.
+    :param data: Data to get IOC type from
+    :return: IOC type if found, None otherwise
+    """
+    data_type = data["type"]
+    return IOC_TYPES.get(data_type.upper(), None)
+
+
+def get_threat_type(data: dict) -> str | None:
+    """
+    Get valid threat type for Sentinel from data.
+    :param data: Data to get threat type from
+    :return: Threat type if found, None otherwise
+    """
+    threat_type = "WatchList"
+    labels = OpenCTIConnectorHelper.get_attribute_in_extension("labels", data)
+    if labels is not None:
+        for label in labels:
+            threat_type = THREAT_TYPES.get(label.upper(), threat_type)
+    return threat_type
+
+
+def get_description(data: dict) -> str:
+    """
+    Get a description according to observable.
+    :param data: Observable data to extract description from
+    :return: Observable description summary or "No Description"
+    """
+    stix_description = OpenCTIConnectorHelper.get_attribute_in_extension(
+        "description", data
+    )
+    return stix_description[0:99] if stix_description is not None else "No description"
+
+
+def get_action(data: dict) -> str:
+    """
+    Get an action according to observable score.
+    :param data: Observable data to get action from
+    :return: Action name or "unknown"
+    """
+    score = OpenCTIConnectorHelper.get_attribute_in_extension("score", data)
+    if score is None:
+        action = "unknown"
+    elif score >= 50:  # self.config.confidence_levek == 50
+        action = "block"
+    elif score < 50 and score != 0:  # self.config.confidence_levek == 50
+        action = "alert"
+    elif score == 0:
+        action = "allow"
+    else:
+        action = "unknown"
+    return action
+
+
+def get_expiration_datetime(data: dict, expiration_time: int) -> str:
+    """
+    Get an expiration datetime for an observable.
+    :param data: Observable data to calculate expiration with
+    :param expiration_time: Duration after which observable is considered as expired
+    :return: Datetime of observable expiration
+    """
+    updated_at = OpenCTIConnectorHelper.get_attribute_in_extension("updated_at", data)
+    datetime_object = datetime.strptime(updated_at, "%Y-%m-%dT%H:%M:%S.%fZ")
+    age = timedelta(expiration_time)
+    expire_datetime = datetime_object + age
+    expiration_datetime = expire_datetime.strftime("%Y-%m-%dT%H:%M:%SZ")
+    return expiration_datetime
+
+
+def get_tlp_level(data: dict) -> str:
+    """
+    Get a TLP level for an observable.
+    :param data: Observable data to extract TLP level from
+    :return: TLP level or "unknown"
```

Comment:
You could perhaps use the message "model" to access the value you are looking for, rather than "jsonify" the whole message to a string.

i.e. `"toto" ==  data["my key"]` rather than `"toto" in str(data) `

Reply:
Done 👍I leveraged this comment to use TLP constants from stix2 instead of "magic" strings

Reply:
Great work !